framework/src/meshgenerators/PatternedMeshGenerator.C

```diff
@@ -128,6 +145,26 @@ PatternedMeshGenerator::generate()
   if (_z_width == 0)
     _z_width = bbox.max()(2) - bbox.min()(2);
 
+  std::vector<boundary_id_type> stitch_bids;
+  if (have_common_ids)
+    stitch_bids = input_bids[0];
+  else
+    for (boundary_id_type id = 0; id != Moose::INVALID_BOUNDARY_ID; ++id)
+      if (!used_boundary_ids.count(id))
+      {
+        stitch_bids.push_back(id);
+        if (stitch_bids.size() == 4)
+          break;
+      }
+
+  for (const auto i : index_range(_meshes))
+    for (const auto side : make_range(4))
+      MeshTools::Modification::change_boundary_id(
+          *_meshes[i], input_bids[i][side], stitch_bids[side]);
```

Comment:
```suggestion
  if (!have_common_ids)
    for (const auto i : index_range(_meshes))
      for (const auto side : make_range(4))
        MeshTools::Modification::change_boundary_id(
            *_meshes[i], input_bids[i][side], stitch_bids[side]);
```

This makes sense right?

Reply:
Yup, will incorporate.